mbs_results/imputation_flags.py

```diff
@@ -2,28 +2,86 @@
 import pandas as pd
 
 
-def create_impute_flags(
+def generate_imputation_marker(
     df: pd.DataFrame,
     target: str,
     period: str,
     reference: str,
     strata: str,
     auxiliary: str,
-    predictive_auxiliary: str,
-    **kwargs
+    time_difference=1,
+    **kwargs,
+) -> pd.DataFrame:
+    """
+    Function to add column containing the a string indicating the method of
+    imputation to use following the hierarchy in specifications
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        DataFrame containing target, auxiliary and manual construction columns
+        and containing forward, backward predictive period columns (
+        These columns are created by calling flag_matched_pair forward
+        and backwards)
+    target : str
+        Column name containing target variable.
+    period: str
+        Column name containing date variable.
+    reference : str
+        Column name containing business reference id.
+    strata : str
+        Column name containing strata information (sic).
+    auxiliary : str
+        Column name containing auxiliary data.
+    time_difference: int
+        lookup distance for matched pairs
+    kwargs : mapping, optional
+        A dictionary of keyword arguments passed into func.
+
+    Returns
+    -------
+    pd.DataFrame
+        Dataframe with additional column containing imputation marker
+        i.e. the type of imputation method that should be used to fill
+        missing returns.
+    """
+    if f"{target}_man" in df.columns:
+        flags = ["r", "mc", "fir", "bir", "fimc", "fic", "c"]
+        # Check order from Specs
+    else:
+        flags = ["r", "fir", "bir", "fic", "c"]
+
+    create_imputation_logical_columns(
+        df, target, period, reference, strata, auxiliary, time_difference
+    )
+
+    select_cols = [f"{i}_flag_{target}" for i in flags]
+    first_condition_met = [np.where(i)[0][0] for i in df[select_cols].values]
+    df[f"imputation_flags_{target}"] = [flags[i] for i in first_condition_met]
+    df.drop(columns=select_cols, inplace=True)
+
+    return df
+
+
+def create_imputation_logical_columns(
```

Comment:
To my understanding (hope i didn't miss anything) that this function is in reverse priority ? Like `R` is first if not na, then `MC` etc , hence should we add a comment about this ? It will help if functions needs extending

Reply:
This function doesnt actually need the order, the order is sorted in `generate_imputation_marker`. I've added extra information to both docstrings to explain this for future reference